### PR TITLE
:sparkles::microscope::snowflake::hammer: Allow user to supply additional functions

### PR DESCRIPTION
**Pull request by bbody**

### CODE REVIEW

File: CMD-RESUME-DATA-SCHEMA.md [hunk 1]
```diff
@@ -68,6 +68,141 @@ Outside of the [JSON Resume](https://jsonresume.org/) specifications some additi
 		"  |                                    |          /___\\/",
 		"  |hectoras                            |           oo",
 		"  |____________________________________|"
-	]
+	],
+	"spiritanimal": "Cassowary",
+    "countriestravelledto": [
```
Comment (bbody) [inline CMD-RESUME-DATA-SCHEMA.md]: Ignore this lol

File: spec/helpers/formatters.spec.js [hunk 2]
```diff
@@ -172,7 +172,7 @@ describe("formatters", function() {
 		});
 
 		it("Doesn\"t override any styles", function() {
-			expect(initStyles(this.defaultStyles, {})).toEqual(this.defaultStyles);
+			expect(initStyles(this.defaultStyles)).toEqual(this.defaultStyles);
```
Comment (obelic2) [inline spec/helpers/formatters.spec.js]: Seems like a superfluous test now

File: js/helpers/commandHandlers.js [hunk 3]
```diff
@@ -175,3 +192,72 @@ var buildSocialNetwork = function(value) {
 	}
 	return value.url ? value.url : false;
 };
+
+var isValidCommandType = function(commandType) {
+	return !!commandType && commandProcessor.hasOwnProperty(commandType);
```
[the remark this answers is not in the document]
Comment (bbody) [reply js/helpers/commandHandlers.js]: It is for transforming things to a boolean
!!(falsy) => !(true) => false
!!(truthy) => !(false) => true

File: spec/helpers/formatters.spec.js (hunk 2 above)
Comment (bbody) [reply spec/helpers/formatters.spec.js]: Probably for code coverage, I will fix that " though

File: js/examples/example-script.js [hunk 4]
```diff
@@ -15,7 +15,67 @@ $(document).ready(function() {
         name: {
             color: "purple"
         },
-        extraDetails: "responses/extra-details.json"
+        extraDetails: "responses/extra-details.json",
+        customCommands: [
```
[the remark this answers is not in the document]
Comment (bbody) [reply js/examples/example-script.js]: I'm assuming you mean the markdown? If so, that is compiled into it. Still a single source of truth.

**Pull request merged**